src/metatensor/models/utils/data/dataset.py

```diff
@@ -93,3 +135,73 @@ def collate_fn(batch):
         )
 
     return structures, targets
+
+
+def check_datasets(
+    train_datasets: List[Dataset],
+    validation_datasets: List[Dataset],
```

Comment:
This can also be a subset.
```suggestion
    validation_datasets: Union[List[Dataset], List[torch.utils.data.Subset]],

```